src/main/java/org/gaul/modernizer_maven_plugin/ModernizerMojo.java

```diff
@@ -144,15 +144,25 @@
      * java/lang/.*
      */
     @Parameter
-    private Set<String> exclusionPatterns = new HashSet<String>();
+    private final Set<String> exclusionPatterns = new HashSet<String>();
```

Comment:
Why do you mark these as `final`?  The Mojo will set these via reflection and this could confuse the JVM.

Reply:
Duh, sorry, that was not intentional; my IDE is configured to do this as an automatic Save Action - sorry. (And it's of course too dumb to understand that _the Mojo will set these via reflection._) How would you feel if I made these `protected` instead of `private`? That will prevent this auto save thing (which is popular in at least 2 major Java IDEs...) from adding the `final`. De facto these fields really aren't all that `private` anyway, agreed? I mean they are part of the Mojo public surface, they are directly referenced by name in pom.xml files.